### PR TITLE
Iteration 4 tt shot accuracy

### PR DESCRIPTION
**Pull request by tylertomlinson**
Most accurate and least accurate team methods created 
Helper method shots_and_goals_per_team 
passes in a season id instead of using team_id, which reduces time on rspec.

Tests created for every method 
Rake passes all tests 
SimpleCov covering over 99%

### CODE REVIEW

File: lib/season_stats.rb [hunk 1]
```diff
@@ -6,9 +6,12 @@ def initialize(games, teams, game_teams)
     super(games, teams, game_teams)
   end
 
-  def shot_accuracy_by_team_id(team_id)
-    (total_goals_by_team_id(team_id).to_f/total_shots_by_team_id(team_id) * 100.0)
-    .round(2)
+  def most_accurate_team(season_id)
+    find_name(shots_and_goals_per_team(season_id).max_by { |team, ratio| ratio}.first)
```
Comment (alex-latham) [inline lib/season_stats.rb:10]: Good word (ratio) to describe what the BV is. I'll implement it in my similar methods, too.

File: lib/season_stats.rb [hunk 2]
```diff
@@ -129,4 +130,18 @@ def worst_coach(season_param)
...
+  def shots_and_goals_per_team(season_id)
+    team_accuracy = {}
+    team_shots = Hash.new(0)
+    team_goals = Hash.new(0)
+    season_game_teams(season_id).each do |game|
+      team_shots[game.team_id] += game.shots
+      team_goals[game.team_id] += game.goals
+    end
+    team_goals.each do |team, goals|
+      team_accuracy[team] = (goals.to_f / team_shots[team])
```
Comment (alex-latham) [inline lib/season_stats.rb:143]: Elegant

Comment (Maxwell-Baird) [reply lib/season_stats.rb:143]: Nice helper method

Comment (whitneykidd) [reply lib/season_stats.rb:143]: Love to see it

**Pull request merged**